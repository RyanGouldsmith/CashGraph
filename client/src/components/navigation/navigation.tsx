import React from 'react';
import { Link } from 'react-router-dom';

interface NavigationProps {
  routeForDashboard: string;
  routeForSpendingCreation: string;
}

export const Navigation = React.memo<NavigationProps>(
  ({ routeForDashboard, routeForSpendingCreation }) => (
    <header>
      <nav>
        <Link to={routeForDashboard}>Home</Link>
        <Link to={routeForSpendingCreation}>Create Spending</Link>
      </nav>
    </header>
  ),
);
