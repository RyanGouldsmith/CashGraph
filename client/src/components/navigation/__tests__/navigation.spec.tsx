import 'babel-polyfill';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { MemoryRouter } from 'react-router';

import { Navigation } from '../navigation';

afterEach(cleanup);

test('should match the snapshot for navigation', async () => {
  const { container } = render(
    <MemoryRouter>
      <Navigation
        routeForDashboard="/knownRouteForDashboard"
        routeForSpendingCreation="/knownRouteForSpendingCreation"
      />
    </MemoryRouter>,
  );

  expect(container).toMatchSnapshot();
});
