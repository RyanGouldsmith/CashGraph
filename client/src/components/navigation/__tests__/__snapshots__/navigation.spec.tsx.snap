// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for navigation 1`] = `
<div>
  <header>
    <nav>
      <a
        href="/knownRouteForDashboard"
      >
        Home
      </a>
      <a
        href="/knownRouteForSpendingCreation"
      >
        Create Spending
      </a>
    </nav>
  </header>
</div>
`;
